Automattic/jetpack
# Add volta configuration for managing node versions

#### Changes proposed in this Pull Request:

This adds a `volta` entries for all package.json files, pinning the node version for those using [Volta](https://volta.sh/).

Some of the benefits over nvm include

- No need to run `nvm use` to select the correct node version.
- When the node version is updated, it's installed automatically.
- Globally installed commands (e.g. `yarn`) are pinned correctly to the project version, without the need for using `npx`.
- Avoids issues that nvm has with alternative shell environments, such as running npm/yarn commands through VS Code's shell.

Volta can replace nvm, but those who wish to continuing using nvm can. As long as the project provides an .nvmrc file, `nvm use` overrides volta, in my testing.

[Related wp-calypso PR](https://github.com/Automattic/wp-calypso/pull/48734).

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* [Install volta](https://docs.volta.sh/guide/getting-started) (`brew install volta` will do it on macOS with homebrew)
* In a directory outside the Jetpack repo, add node and yarn to your volta toolchain, e.g.

```
$ volta install node@latest
success: installed and set node@16.0.0 (with npm@7.10.0) as default
$ volta install yarn
success: installed and set yarn@1.22.10 as default
$ volta list
⚡️ Currently active tools:

    Node: v16.0.0 (default)
    Yarn: v1.22.10 (default)
```

* Now switch to the jetpack repo directory with this branch checked out and run `volta list`, you should see the versions specified in package.json

```
$ volta list
⚡️ Currently active tools:

    Node: v14.16.0 (current @ .../jetpack/package.json)
    Yarn: v1.22.10 (current @ .../jetpack/package.json)
```

- And the same for all of the packages in the repo

```
$ cd projects/plugins/jetpack 
$ volta list
⚡️ Currently active tools:

    Node: v14.16.0 (current @
    .../jetpack/projects/plugins/jetpack/package.json)
    Yarn: v1.22.10 (current @
    .../jetpack/projects/plugins/jetpack/package.json)
```
